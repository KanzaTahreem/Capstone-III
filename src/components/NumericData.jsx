import React from 'react';
import { useLocation } from 'react-router-dom';

const NumericData = () => {
  const location = useLocation();
  const { coin } = location.state;

  return (
    <>
      <div>{coin.id}</div>
      <div>
        <img src={coin.image} alt={`${coin.image}-logo`} />
        <p>
          {coin.name}
          /
          {coin.symbol}
        </p>
      </div>
      <div>
        <p>{`$${coin.current_price}`}</p>
        <p>
          market-cap:
          {' '}
          {coin.market_cap}
        </p>
        <p>
          market cap change:
          {' '}
          {coin.market_cap_change_24h}
        </p>
        <p>
          market change %:
          {' '}
          {coin.market_cap_change_percentage}
        </p>
        <p>
          rank:
          {' '}
          {coin.market_cap_rank}
        </p>
        <p>
          dilutated value:
          {' '}
          {coin.fully_diluted_valuation}
        </p>

        <p>
          high 24h:
          {' '}
          {coin.high_24h}
        </p>
        <p>
          low 24h:
          {' '}
          {coin.low_24h}
        </p>
        <p>
          price change 24h:
          {' '}
          {coin.price_change_24h}
        </p>
        <p>
          price change %:
          {' '}
          {coin.price_change_percentage_24h}
        </p>
        <p>
          total supply:
          {' '}
          {coin.total_supply}
        </p>
        <p>
          max supply:
          {' '}
          {coin.max_supply}
        </p>
        <p>
          ath:
          {' '}
          {coin.ath}
        </p>
        <p>
          ath change %:
          {' '}
          {coin.ath_change_percentage}
        </p>
        <p>
          last updated:
          {' '}
          {coin.last_updated}
        </p>
      </div>
    </>
  );
};

export default NumericData;
