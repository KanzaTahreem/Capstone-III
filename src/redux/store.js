import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './AllCoins/coins';
import chartReducer from './CoinChart/chart';

const store = configureStore({
  reducer: {
    crypto: coinsReducer,
    chart: chartReducer,
  },
});

export default store;
