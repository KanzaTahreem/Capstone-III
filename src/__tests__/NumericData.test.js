import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import NumericData from '../components/NumericData';

it('renders correctly', () => {
  const rocket = renderer
    .create(
      <Router>
        <Provider store={store}>
          <NumericData />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(rocket).toMatchSnapshot();
});
