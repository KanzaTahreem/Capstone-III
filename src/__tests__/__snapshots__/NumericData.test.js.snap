// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="coin-details"
  data-testid="coin-stats"
>
  <div
    className="coin"
  >
    <div>
      <p
        className="name"
      >
         
        (undefined)
      </p>
      <img
        alt="undefined-logo"
      />
    </div>
    <div>
      <div
        className="price"
      >
        <p>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
              />
              <path
                d="M12 11c-2 0-2-.63-2-1s.7-1 2-1 1.39.64 1.4 1h2A3 3 0 0 0 13 7.12V6h-2v1.09C9 7.42 8 8.71 8 10c0 1.12.52 3 4 3 2 0 2 .68 2 1s-.62 1-2 1c-1.84 0-2-.86-2-1H8c0 .92.66 2.55 3 2.92V18h2v-1.08c2-.34 3-1.63 3-2.92 0-1.12-.52-3-4-3z"
              />
            </svg>
          </span>
          <span>
            Price
          </span>
        </p>
        <p>
          $ undefined
        </p>
      </div>
      <div
        className="rank"
      >
        <p>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.39 12.648a1.32 1.32 0 0 0-.015.18c0 .305.21.508.5.508.266 0 .492-.172.555-.477l.554-2.703h1.204c.421 0 .617-.234.617-.547 0-.312-.188-.53-.617-.53h-.985l.516-2.524h1.265c.43 0 .618-.227.618-.547 0-.313-.188-.524-.618-.524h-1.046l.476-2.304a1.06 1.06 0 0 0 .016-.164.51.51 0 0 0-.516-.516.54.54 0 0 0-.539.43l-.523 2.554H7.617l.477-2.304c.008-.04.015-.118.015-.164a.512.512 0 0 0-.523-.516.539.539 0 0 0-.531.43L6.53 5.484H5.414c-.43 0-.617.22-.617.532 0 .312.187.539.617.539h.906l-.515 2.523H4.609c-.421 0-.609.219-.609.531 0 .313.188.547.61.547h.976l-.516 2.492c-.008.04-.015.125-.015.18 0 .305.21.508.5.508.265 0 .492-.172.554-.477l.555-2.703h2.242l-.515 2.492zm-1-6.109h2.266l-.515 2.563H6.859l.532-2.563z"
              />
            </svg>
          </span>
          <span>
            Rank
          </span>
        </p>
        <p />
      </div>
    </div>
  </div>
  <h2
    className="stats-title"
  >
     
    (undefined)
     
    Statistics
  </h2>
  <div
    className="coin-stats"
  >
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463 20.3L315.6 75.65 368.5 109C274 228 137.3 301.3 21.8 329.2l32.72 59.6C184.3 346.5 321.7 270.2 410.2 135.4l40.3 25.4L463 20.3zm7.7 116.7l-4.8 54.8-51.3-32.4c-1.8 2.7-3.7 5.3-5.6 8V487h78V137h-16.3zM320.1 265c-12.7 11-25.8 21.4-39.1 31.2V487h78V265h-38.9zm-123.7 84c-14.4 7.6-28.8 14.6-43.4 21.2V487h78V349h-34.6zM25 393v94h78v-94h-5.68c-14.82 5.5-29.63 10.6-44.35 15.3l-7.06 2.2-9.6-17.5H25z"
            />
          </svg>
        </span>
        <span>
           Market Cap
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5.373 4.51A9.962 9.962 0 0 1 12 2c5.523 0 10 4.477 10 10a9.954 9.954 0 0 1-1.793 5.715L17.5 12H20A8 8 0 0 0 6.274 6.413l-.9-1.902zm13.254 14.98A9.962 9.962 0 0 1 12 22C6.477 22 2 17.523 2 12c0-2.125.663-4.095 1.793-5.715L6.5 12H4a8 8 0 0 0 13.726 5.587l.9 1.902zm-5.213-4.662L10.586 12l-2.829 2.828-1.414-1.414 4.243-4.242L13.414 12l2.829-2.828 1.414 1.414-4.243 4.242z"
              />
            </g>
          </svg>
        </span>
        <span>
          Market Cap Change
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
         / 
        undefined %
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 22c3.976 0 8-1.374 8-4V6c0-2.626-4.024-4-8-4S4 3.374 4 6v12c0 2.626 4.024 4 8 4zm0-2c-3.722 0-6-1.295-6-2v-1.268C7.541 17.57 9.777 18 12 18s4.459-.43 6-1.268V18c0 .705-2.278 2-6 2zm0-16c3.722 0 6 1.295 6 2s-2.278 2-6 2-6-1.295-6-2 2.278-2 6-2zM6 8.732C7.541 9.57 9.777 10 12 10s4.459-.43 6-1.268V10c0 .705-2.278 2-6 2s-6-1.295-6-2V8.732zm0 4C7.541 13.57 9.777 14 12 14s4.459-.43 6-1.268V14c0 .705-2.278 2-6 2s-6-1.295-6-2v-1.268z"
            />
          </svg>
        </span>
        <span>
          Dilutated Value
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.354 4.354a.5.5 0 0 0-.708-.708L5 10.293 1.854 7.146a.5.5 0 1 0-.708.708l3.5 3.5a.5.5 0 0 0 .708 0l7-7zm-4.208 7-.896-.897.707-.707.543.543 6.646-6.647a.5.5 0 0 1 .708.708l-7 7a.5.5 0 0 1-.708 0z"
            />
            <path
              d="m5.354 7.146.896.897-.707.707-.897-.896a.5.5 0 1 1 .708-.708z"
            />
          </svg>
        </span>
        <span>
          Total Supply
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M5.293 7.707a1 1 0 010-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 01-1.414 1.414L11 5.414V17a1 1 0 11-2 0V5.414L6.707 7.707a1 1 0 01-1.414 0z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span>
          24h High
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M14.707 12.293a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 111.414-1.414L9 14.586V3a1 1 0 012 0v11.586l2.293-2.293a1 1 0 011.414 0z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span>
          24h Low
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 13h16v2h-16zM2 9h2v3h-2zM5 5h2v7h-2zM8 8h2v4h-2zM11 2h2v10h-2z"
            />
          </svg>
        </span>
        <span>
          24h Price Change
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
         / 
        undefined %
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h1v15h15v1H0V0zm10 3.5a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-1 0V4.9l-3.613 4.417a.5.5 0 0 1-.74.037L7.06 6.767l-3.656 5.027a.5.5 0 0 1-.808-.588l4-5.5a.5.5 0 0 1 .758-.06l2.609 2.61L13.445 4H10.5a.5.5 0 0 1-.5-.5z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span>
          All Time High
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
         / 
        undefined %
      </p>
    </div>
    <div>
      <p
        className="stats-name"
      >
        <span>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h1v15h15v1H0V0zm10 11.5a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 .5-.5v-4a.5.5 0 0 0-1 0v2.6l-3.613-4.417a.5.5 0 0 0-.74-.037L7.06 8.233 3.404 3.206a.5.5 0 0 0-.808.588l4 5.5a.5.5 0 0 0 .758.06l2.609-2.61L13.445 11H10.5a.5.5 0 0 0-.5.5z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span>
          All Time Low
        </span>
      </p>
      <p
        className="value"
      >
        $ undefined
         / 
        undefined %
      </p>
    </div>
  </div>
</div>
`;
